syntax = "proto2";

package NCloud.NBlockStore.NProto;

option go_package = "github.com/ydb-platform/nbs/cloud/blockstore/config";

////////////////////////////////////////////////////////////////////////////////

message TLogbrokerConfig
{
    enum EProtocol
    {
        PROTOCOL_UNSPECIFIED = 0;
        PROTOCOL_PQ0 = 1;
        PROTOCOL_TOPIC_API = 2;
    };

    message TIamJwtFile
    {
        optional string IamEndpoint = 1;
        optional string JwtFilename = 2;
    }

    message TIamMetadataServer
    {
        optional string Endpoint = 1;
    };

    optional string Address = 1;
    optional uint32 Port = 2;
    optional string Database = 3;
    optional bool UseLogbrokerCDS = 4;
    optional string CaCertFilename = 5;

    optional string Topic = 6;
    optional string SourceId = 7;

    // use AuthConfig.IamMetadataServer instead
    optional string MetadataServerAddress = 8 [deprecated = true];

    optional EProtocol Protocol = 9;

    oneof AuthConfig {
        TIamJwtFile IamJwtFile = 10;
        TIamMetadataServer IamMetadataServer = 11;
    };
}
